src/stratis_cli/_actions/_data.py

```diff
@@ -205,31 +207,76 @@
 _MANAGER_INTERFACE = "org.storage.stratis1.Manager"
 
 DBUS_TIMEOUT_SECONDS = 120
+MAXIMUM_DBUS_TIMEOUT_MS = 1073741823
+
+
+def _get_timeout(value):
+    """
+    Turn an input str or int into a float timeout value.
+
+    :param value: the input str or int
+    :type value: str or int
+    :raises StratisCliEnvironmentError:
+    :returns: float
+    """
+    # Ensure the input str is not a float
+    if isinstance(value, float):
+        raise StratisCliEnvironmentError(
+            "The timeout value you provided is a float; it should be an integer."
+        )
+
+    try:
+        timeout_int = int(value)
+
+    except ValueError:
+        raise StratisCliEnvironmentError(
+            "The timeout value you provided is not an integer."
+        )
+
+    # Ensure the integer is not too small
+    if timeout_int < -1:
+        raise StratisCliEnvironmentError(
+            "The timeout value you provided is smaller than the smallest acceptable value, -1."
+        )
+
+    # Ensure the integer is not too large
+    if timeout_int > MAXIMUM_DBUS_TIMEOUT_MS:
+        raise StratisCliEnvironmentError(
+            "The timeout value you provided exceeds the largest acceptable value, "
```

Comment:
Use string formatting here:

```The timeout value...largest acceptable value, %s." % MAXIMUM_DBUS_TIMEOUT_MS```

it'll apply the string formatting automatically, since you're using ```%s```.